diff --git a/node_modules/eml-parse-js/dist/index.es.js b/node_modules/eml-parse-js/dist/index.es.js
index e36270c..5c4d2c2 100644
--- a/node_modules/eml-parse-js/dist/index.es.js
+++ b/node_modules/eml-parse-js/dist/index.es.js
@@ -1823,9 +1823,11 @@ function getBoundary(contentType) {
   const match = /(?:B|b)oundary=(?:'|")?(.+?)(?:'|")?(\s*;[\s\S]*)?$/g.exec(contentType);
   return match ? match[1] : void 0;
 }
+/* disused
 function getCharsetName(charset) {
   return charset.toLowerCase().replace(/[^0-9a-z]/g, "");
 }
+*/
 function guid() {
   return "xxxxxxxxxxxx-4xxx-yxxx-xxxxxxxxxxxx".replace(/[xy]/g, function(c2) {
     const r = Math.random() * 16 | 0, v = c2 == "x" ? r : r & 3 | 8;
@@ -1854,6 +1856,7 @@ function mimeDecode(str = "", fromCharset = "UTF-8") {
   }
   return decode(buffer, fromCharset);
 }
+/* disused
 const GB2312UTF8 = {
   Dig2Dec: function(s) {
     let retV = 0;
@@ -1975,6 +1978,7 @@ const GB2312UTF8 = {
     return substr + str1;
   }
 };
+*/
 function _handleAddress(tokens) {
   let token;
   let isGroup = false;
@@ -2259,19 +2263,29 @@ function getEmailAddress(rawStr) {
   return list;
 }
 function decodeJoint(str) {
+  // add regex for file name
+  const nameRegex = /\?(B|Q)\?.+?={0,2}\?/gi;
+  const matchName = str.match(nameRegex);
   const match = /=\?([^?]+)\?(B|Q)\?(.+?)(\?=)/gi.exec(str);
-  if (match) {
-    const charset = getCharsetName(match[1] || defaultCharset);
+  if (match && matchName) {    
+    const charset = match[1] || defaultCharset;// remove "getCharsetName" function
     const type = match[2].toUpperCase();
-    const value = match[3];
+    const values = matchName.map((value) => {
+      var _a;
+      const groups = (_a = /\?(B|Q)\?(?<name>.+?={0,2})\?/gi.exec(value)) == null ? void 0 : _a.groups;
+      if (groups == null ? void 0 : groups.name) {
+        return groups.name;
+      }
+      return "";
+    });
     if (type === "B") {
       if (charset === "utf8") {
-        return decode(encode(Base64.fromBase64(value.replace(/\r?\n/g, ""))), "utf8");
+        return values.map((value) => decode(encode(Base64.fromBase64(value.replace(/\r?\n/g, ""))), "utf8")).join("");        
       } else {
-        return decode(Base64.toUint8Array(value.replace(/\r?\n/g, "")), charset);
+        return values.map((value) => decode(Base64.toUint8Array(value.replace(/\r?\n/g, "")), charset)).join("");        
       }
     } else if (type === "Q") {
-      return unquotePrintable(value, charset, true);
+      return values.map((value) => unquotePrintable(value, charset, true)).join("");
     }
   }
   return str;
@@ -2634,14 +2648,14 @@ function read(eml, options, callback) {
   function _append(headers, content, result2) {
     const contentType = headers["Content-Type"] || headers["Content-type"];
     const contentDisposition = headers["Content-Disposition"];    
-    const charset = getCharsetName(getCharset(contentType) || defaultCharset);
+    const charset = getCharset(contentType) || defaultCharset;// remove "getCharsetName function"
     let encoding2 = headers["Content-Transfer-Encoding"] || headers["Content-transfer-encoding"];
     if (typeof encoding2 === "string") {
       encoding2 = encoding2.toLowerCase();
     }
     if (encoding2 === "base64") {
       if (contentType && contentType.indexOf("gbk") >= 0) {
-        content = encode(GB2312UTF8.GB2312ToUTF8(content.replace(/\r?\n/g, "")));
+        content = encode(content.replace(/\r?\n/g, ""), "gbk"); // remove "GB2312ToUTF8" object
       } else {
         content = encode(content.replace(/\r?\n/g, ""));
       }
@@ -2654,11 +2668,11 @@ function read(eml, options, callback) {
       if (typeof content !== "string") {
         content = decode(content, charset);
       }
-      let htmlContent = content.replace(/\r\n|(&quot;)/g, "").replace(/\"/g, `"`);
+      let htmlContent = content.replace(/\r\n|(&quot;)/g, "");
       try {        
         if (encoding2 === "base64") {
           htmlContent = Base64.decode(htmlContent);          
-        } else if (Base64.btoa(Base64.atob(htmlContent)) == htmlContent) {
+        } else if (Base64.isValid(htmlContent)) { // remove Base64.btoa(Base64.atob(htmlContent)) == htmlContent          
           htmlContent = Base64.atob(htmlContent);          
         }
       } catch (error2) {
@@ -2703,7 +2717,7 @@ function read(eml, options, callback) {
       for (const key of NameContainer) {
         const name = headers[key];
         if (name) {
-          result_name = name.replace(/(\s|'|utf-8|\*[0-9]\*)/g, "").split(";").map((v) => /name[\*]?="?(.+?)"?$/gi.exec(v)).reduce((a, b) => {
+          result_name = name.replace(/(\s|'|\*[0-9]\*)/g, "").split(";").map((v) => /name[\*]?="?(.+?)"?$/gi.exec(v)).reduce((a, b) => {
             if (b && b[1]) {
               a += b[1];
             }
@@ -2715,7 +2729,7 @@ function read(eml, options, callback) {
         }
       }
       if (result_name) {
-        attachment.name = decodeURI(result_name);
+        attachment.name = decodeJoint(result_name);// modify decodeURI to decodeJoint
       }
       const ct = headers["Content-Type"] || headers["Content-type"];
       if (ct) {
@@ -2844,7 +2858,7 @@ function read(eml, options, callback) {
 }
 export {
   Base642 as Base64,
-  GB2312UTF8 as GBKUTF8,
+  //GB2312UTF8 as GBKUTF8,
   build as buildEml,
   completeBoundary,
   convert,
